lib/puppet_x/puppetlabs/powershell/powershell_manager.rb

```diff
@@ -64,9 +64,14 @@ def initialize(cmd, debug)
             sleep 0.2
           end
         end
-
-        fail "Failure waiting for PowerShell process #{@ps_process[:pid]} to start pipe server" if @pipe.nil?
-
+        if @pipe.nil?
+          # Tear down and kill the process if unable to connect to the pipe; failure to do so
+          # results in zombie processes being left after the puppet run.
+          @stdout.close if !@stdout.closed?
+          @stderr.close if !@stderr.closed?
+          Process.kill("KILL", @ps_process[:pid])
```

Comment:
So after verifying while pairing, this seems correct.

One thing that we do need to change slightly is to make:

```ruby
Process.kill("KILL", @ps_process[:pid]) if @ps_process.alive?
```

This ensures that the waiter thread is still up and monitoring the child process, which is a proxy for the child process still running.  If the waiter thread is *dead*, there's no guarantee the PID is even valid, or that another process hasn't claimed it.

We want to be careful about introducing the "PowerShell module chaos monkey" on unsuspecting victims.